UnitsNet.Serialization.JsonNet/UnitsNetBaseJsonConverter.cs

```diff
@@ -87,7 +116,40 @@ protected IQuantity ConvertValueUnit(ValueUnit valueUnit)
             };
         }
 
-        private static Enum GetUnit(string unit)
+        private Type GetRegisteredType(string unit)
+        {
+            (var unitEnumTypeName, var unitEnumValue) = SplitUnitString(unit);
+            return _registeredTypes.Find(t => t.Unit.Name == unitEnumTypeName).Quantity;
+        }
+
+        private Enum GetUnit(string unit)
+        {
+            (var unitEnumTypeName, var unitEnumValue) = SplitUnitString(unit);
+
+            // First try to find the name in the list of registered types.
+            var unitEnumType = _registeredTypes.Find(t => t.Unit.Name == unitEnumTypeName).Unit;
```

Comment:
It seems like a small weakness in the current design that the namespace is not included in the serialization. But I guess for now, this is what we have to work with.